impl/src/main/java/org/ehcache/impl/serialization/CompactJavaSerializer.java

```diff
@@ -65,7 +66,19 @@
    * @see Serializer
    */
   public CompactJavaSerializer(ClassLoader loader) {
+    this(loader, new TransientStateRepository());
+  }
+
+  public CompactJavaSerializer(ClassLoader loader, StateRepository stateRepository) {
     this.loader = loader;
+    this.readLookup = stateRepository.getPersistentConcurrentMap("readLookup");
```

Comment:
Should we have a name that ties to the FQCN instead of (or maybe in addition to) the field name or purpose here: e.g "CompactJavaSerializer-ObjectStreamClassIndex"